src/front/glsl_new/parser.rs

```diff
@@ -156,18 +156,17 @@ pomelo! {
             }
         } else {
             // try global and local vars
-            if let Some(global_var) = extra.lookup_global_variables.get(&v.1) {
-                ExpressionRule{
-                    expression: extra.context.expressions.append(Expression::GlobalVariable(*global_var)),
-                    statements: vec![],
-                }
-            } else if let Some(local_var) = extra.context.lookup_local_variables.get(&v.1) {
-                ExpressionRule{
-                    expression: extra.context.expressions.append(Expression::LocalVariable(*local_var)),
-                    statements: vec![],
-                }
-            } else {
-                return Err(ErrorKind::UnknownVariable(v.0, v.1))
+            let expr =
+                if let Some(global_var) = extra.lookup_global_variables.get(&v.1) {
```

Comment:
Should we first look the local scopes and then the global vars?

Reply:
of course :D